scheduler/util_test.go

```diff
@@ -551,13 +553,19 @@ func TestInplaceUpdate_Success(t *testing.T) {
 		DesiredStatus: structs.AllocDesiredStatusRun,
 	}
 	alloc.TaskResources = map[string]*structs.Resources{"web": alloc.Resources}
+	alloc.PopulateServiceIDs()
 	noErr(t, state.UpsertAllocs(1001, []*structs.Allocation{alloc}))
 
+	if alloc.Services["web-frontend"] == "" {
```

Comment:
Added a test which proves that a service is removed from the map in alloc when it is removed from a task definition